constraint-solver/src/range_constraint.rs

```diff
@@ -12,7 +13,8 @@ use powdr_number::{log2_exact, FieldElement, LargeInt};
 /// and bit masks. The actual constraint is the conjunction of the two.
 ///
 /// Note that the same constraint can have multiple representations.
-#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash, Serialize, Deserialize)]
+#[serde(bound(deserialize = ""))]
```

Comment:
what does this line do?

Reply:
The derive macro has some logic to find out what the correct trait bounds on generic parameters (like `T` here) are. This heuristic fails and `FieldElement` already requires serialize/deserialize, so everything is fine.